tests/ballerina-compiler-api-test/src/test/resources/test-src/symbols/error_type_symbol_test.bal

```diff
@@ -43,3 +43,23 @@ function test() {
     CancelledError distinctErr;
     IntersectionError intersectionErr;
 }
+
+type ErrorDetail1 record {
+    string msg;
+    int value?;
+};
+
+type ErrorDetail2 record {|
+    *ErrorDetail1;
+    boolean flag;
+|};
+
+type SimpleError error<ErrorDetail1>;
+type InclusionError error<ErrorDetail2>;
+
+type SimpleIntersectionError distinct error & error<ErrorDetail1> & error<ErrorDetail1>;
+type MultipleIntersectionError InclusionError & error<ErrorDetail1> & SimpleError & error<ErrorDetail2>;
+function testMultipleIntersectionError() {
+    SimpleIntersectionError simpleErr;
+    MultipleIntersectionError multipleErr;
+}
```

Comment:
in the original issue, there are union types defined in the error type record details.
shall we add a one test for that scenario?